lib/public-suffix.js

```diff
@@ -12,17 +12,25 @@ var extractTldFromHost = require('./from-host.js');
  * @param {string} hostname
  * @return {string}
  */
-module.exports = function getPublicSuffix(rules, hostname) {
+module.exports = function getPublicSuffix(rules, hostname, allowIcann, allowPrivate) {
```

Comment:
Here, I'd change `allowIcann, allowPrivate` in an `options` object.

At the same time, if it's not part of the public API, it does not matter as much.